mantidimaging/gui/windows/operations/presenter.py

```diff
@@ -401,7 +430,12 @@ def _show_negative_values_error(self, negative_stacks: List[StackVisualiserView]
             for i in range(len(stack.presenter.images.data)):
                 if np.any(stack.presenter.images.data[i] < 0):
                     negative_slices.append(i)
-            getLogger(__name__).error(f"Slices containing negative values in {stack.name}: {negative_slices}")
+            slices_msg = f'Slices containing negative values in {stack.name}: '
+            if len(negative_slices) == len(stack.presenter.images.data):
+                slices_msg += "all slices."
+            else:
+                slices_msg += f'{", ".join(_generate_slices_range_list(negative_slices))}'
+            getLogger(__name__).error(slices_msg)
```

Comment:
Could you use the same message in both the GUI and the log message.

Maybe to avoid the GUI message getting to long something like
```
ranges = _generate_slices_range_list(negative_slices)
if len(ranges) > 12:
  message = "  {", ".join(ranges[:10])} ... ranges[-1]" 
else:
  message = "  {", ".join(ranges)}" 
```